### PR TITLE
OSSM-1872 Deprecate v1.1 from istio-operator 2.3.0

### PR DESCRIPTION
**Pull request by yxun**
Signed-off-by: Yuanlin <yuanlin.xu@redhat.com>

### CODE REVIEW

File: deploy/maistra-operator.yaml [hunk 1]
```diff
@@ -10653,31 +10653,6 @@ spec:
...
-        olm.relatedImage.v1_0.citadel: docker.io/maistra/citadel-ubi8:1.0.8
-        olm.relatedImage.v1_0.cni: docker.io/maistra/istio-cni-ubi8:1.0.8
-        olm.relatedImage.v1_0.galley: docker.io/maistra/galley-ubi8:1.0.8
-        olm.relatedImage.v1_0.grafana: docker.io/maistra/grafana-ubi8:1.0.8
-        olm.relatedImage.v1_0.mixer: docker.io/maistra/mixer-ubi8:1.0.8
-        olm.relatedImage.v1_0.pilot: docker.io/maistra/pilot-ubi8:1.0.8
-        olm.relatedImage.v1_0.prometheus: docker.io/maistra/prometheus-ubi8:1.0.8
-        olm.relatedImage.v1_0.proxy-init: docker.io/maistra/proxy-init-centos7:1.0.8
-        olm.relatedImage.v1_0.proxyv2: docker.io/maistra/proxyv2-ubi8:1.0.8
-        olm.relatedImage.v1_0.sidecar-injector: docker.io/maistra/sidecar-injector-ubi8:1.0.8
```
Comment (jewertow) [inline deploy/maistra-operator.yaml:10684]: As far as I remember correctly, we didn't remove these images for 1.0 intentionally, so I guess we shouldn't remove them and the images for 1.1 as well.

Comment (jewertow) [reply deploy/maistra-operator.yaml:10684]: @jwendell could you confirm?

**Comment by luksa** [comment]
We should also remove the 1.0 container from the CNI daemonset.

**Comment by luksa** [comment]
Oh, and the 1.0 strategy.

**Comment by yxun** [comment]
This PR is WIP. I converted it back for testing the new OpenShift CI jobs

**Comment by yxun** [comment]
> 

@luksa I removed the CNI v1.1 daemonset in both overlay and 2.3 templates. Is that correct or should I not remove the overlay one ?

**Comment by yxun** [comment]
> Oh, and the 1.0 strategy.

We don't have 1.0 strategy in maistra-2.3 branch now. However, we have 1.1 strategy.
When I tried to remove that 1.1 strategies, I got some test failures and controller versions need additional changes to align with the strategy removal. 

Do we need to remove the v1_1 strategy in 2.3.0 ?

**Comment by yxun** [comment]
I removed some v1.1 related test cases in this PR. Let me know if we still need to keep some v1.1 test cases or I missed any other tests. Thanks,

File: deploy/maistra-operator.yaml (hunk 1 above)
Comment (yxun) [reply deploy/maistra-operator.yaml:10684]: @jewertow I haven't found a mail or doc about keeping those annotations. Do we still need them ?

**Comment by dgn** [comment]
I wonder if we can remove the SMCPv1 API at some point

File: deploy/maistra-operator.yaml (hunk 1 above)
Comment (dgn) [reply deploy/maistra-operator.yaml:10684]: not sure it was intentional. let's remove both

**Comment by luksa** [comment]
You should also probably update the min version here: https://github.com/Maistra/istio-operator/blob/78ef39ec9034ff0139f4d04e51152ae0e36d6280/pkg/version/version.go#L17

**Review by jwendell — APPROVED** [review]

**Comment by luksa** [comment]
There are a few more references to v1.1. For example:
- https://github.com/Maistra/istio-operator/blob/15f0a4411687264914209cc9c8bd23abb8ef306d/pkg/controller/versions/versions.go#L56
- https://github.com/Maistra/istio-operator/blob/15f0a4411687264914209cc9c8bd23abb8ef306d/pkg/controller/versions/strategy_v1_1.go
- https://github.com/Maistra/istio-operator/blob/95bca82d023e293a569c650f48fd530d9694e28d/Makefile#L81-L115
- https://github.com/Maistra/istio-operator/blob/36327f8a0e3f832d1c3a02145de16e4aa4cd268c/pkg/bootstrap/cni.go#L48
- ...

I guess we can remove these later.

**Comment by yxun** [comment]
This PR removes 1.1 tests and I added do not merge label to avoid causing other open PR conflicts. When other 2.3 PRs are merged, I can rebase this one later. Thanks.

File: resources/helm/overlays/istio_cni/templates/daemonset.yaml [hunk 2]
```diff
@@ -56,51 +56,6 @@ spec:
       containers:
         # This container installs the Istio CNI binaries
         # and CNI network config file on each node.
-{{- if and (.Values.cni.supportedReleases) (has "v1.1" .Values.cni.supportedReleases) }}
```
Comment (yxun) [inline resources/helm/overlays/istio_cni/templates/daemonset.yaml:59]: Should we keep the v1.1 cni container in the template ?

Comment (luksa) [reply resources/helm/overlays/istio_cni/templates/daemonset.yaml:59]: I think you removed the v1.1 images, right?

Comment (yxun) [reply resources/helm/overlays/istio_cni/templates/daemonset.yaml:59]: right, I was thinking what would happen in the existing cluster.
I am clear now.

**Review by luksa — APPROVED** [review]

**Comment by yxun** [comment]
> There are a few more references to v1.1. For example:
> 
>     * https://github.com/Maistra/istio-operator/blob/15f0a4411687264914209cc9c8bd23abb8ef306d/pkg/controller/versions/versions.go#L56
> 
>     * https://github.com/Maistra/istio-operator/blob/15f0a4411687264914209cc9c8bd23abb8ef306d/pkg/controller/versions/strategy_v1_1.go
> 
>     * https://github.com/Maistra/istio-operator/blob/95bca82d023e293a569c650f48fd530d9694e28d/Makefile#L81-L115
> 
>     * https://github.com/Maistra/istio-operator/blob/36327f8a0e3f832d1c3a02145de16e4aa4cd268c/pkg/bootstrap/cni.go#L48
> 
>     * ...
> 
> 
> I guess we can remove these later.

I updated this PR. The cni v1.1 configmap setting is removed and Makefile v1.1 targets are removed.

I left the Strategy V1_1. I couldn't deprecate that one until cleaning all maistra apis v1 code.
I created a task for that remaining work. We can remove those later. 
https://issues.redhat.com/browse/OSSM-2106

**Review by luksa — APPROVED** [review]

**Pull request merged**